core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -208,9 +208,14 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging {
     //query Cromwell again, and restart from the top of this function with the results.
     workflowsWithOutputsF flatMap { workflowsWithOutputs =>
       datasource.inTransaction { dataAccess =>
-        handleOutputs(workflowsWithOutputs, dataAccess) flatMap { _ =>
-          checkOverallStatus(dataAccess) map {
-            shouldStop => StatusCheckComplete(shouldStop)
+        //re-fetch the workflow records because their version number has been bumped by the batchUpdateStatus
```

Comment:
this isn't related to the issue we're seeing in production, but it does fix a bug (that I've written a test for -- [below](https://github.com/broadinstitute/rawls/pull/673/files#diff-992861fb0cbe05f3ac6b6b8fb9b812caR394)) that I found while trying to repro this issue.